src/VNCVideoCapture.cpp

```diff
@@ -0,0 +1,206 @@
+#include <rfb/rfbclient.h>
+#include <signal.h>
+#include <time.h>
+#include "VNCVideoCapturer.h"
+
+#include "rtc_base/thread.h"
+#include "rtc_base/logging.h"
+#include "api/video_codecs/video_decoder.h"
+#include "media/base/videocapturer.h"
+#include "media/engine/internaldecoderfactory.h"
+#include "api/video/i420_buffer.h"
+#include "libyuv/video_common.h"
+#include "libyuv/convert.h"
+#include <iostream>
+
+char *host = "127.0.0.1";
+int port = 5900;
+
+static char * get_password (rfbClient *client) {
+  VNCVideoCapturer* that = (VNCVideoCapturer *) rfbClientGetClientData(client, (void *) "this");
+  return that->onGetPassword();
+}
+
+static void get_frame(rfbClient* client) {
+  VNCVideoCapturer* that = (VNCVideoCapturer *) rfbClientGetClientData(client, (void *) "this");
+  return that->onFrameBufferUpdate();
+}
+
+static void signal_handler(int sig) {
+  if (sig == SIGINT) {
+	RTC_LOG(LERROR) << __PRETTY_FUNCTION__ << "Recieved SIGINT, exiting ...";
+	exit(EXIT_FAILURE);
+  }
+}
+
+char* VNCVideoCapturer::onGetPassword() {
+  std::cout<< "Starting to get PASSWORD!!!" << std::endl;
+  // remove vnc:// prefix from uri before passing along
+  std::string url = uri.substr(6, std::string::npos);
```

Comment:
I would be tempted to use a "proper" URI parser (it'd make extracting the password look nicer as well), or at least validate that the URI is the scheme we expect.

Reply:
Looked around and could not find a standalone url parsing library that I can easily import in the project. Gonna write a simple one based on regex mentioned in: https://tools.ietf.org/html/rfc3986#page-50